linux_os/guide/system/network/network_implement_access_control/oval/shared.xml

```diff
@@ -0,0 +1,101 @@
+<def-group>
+    <definition class="compliance" id="{{{ rule_id }}}" version="1">
+    {{{ oval_metadata("Configure the access control program to grant or deny access to
+      specific hosts and services") }}}
+    <criteria operator="OR">
+      <criteria operator="AND">
+        <extend_definition comment="package firewalld enabled" definition_ref="package_firewalld_installed" />
+        <extend_definition comment="service firewalld enabled" definition_ref="service_firewalld_enabled" />
+        <criterion comment="firewalld is configured to use the public zone as the default"
+          test_ref="test_firewalld_public_default_zone" />
+        <criteria operator="OR">
+            <criterion comment="firewalld's public zone is configured to grant access to
+                specific hosts" test_ref="test_firewalld_public_zone_hosts_configured" />
+            <criterion comment="firewalld's public zone is configured to grant access to
+                specific services" test_ref="test_firewalld_public_zone_services_configured" />
+            <criterion comment="/etc/firewalld/zones/public does not exist
```

Comment:
From `firewalld.zones(5)`:
> [...] you can create or copy a zone file in one of the configuration directories.  /usr/lib/firewalld/zones is used for default and fallback configurations and /etc/firewalld/zones is used for user created and customized configuration files.

So the logic in the OVAL is to look for configurations for specific hosts or services in a user-defined public zone. Or to make sure that there is no user defined public zone and therefore the one included with the firewalld packages is being used. This works under the assumption (perhaps a **big** assumption) that the default public zone is properly configured and was not modified.

Now that I write this I think that rather than forcing users to configure the `public` zone as the default one. A smarter approach wold be to get whatever zone is configured as the default and then check that it is properly configured; either in `/usr/lib` or in `/etc`.